### DISCUSSION TITLE
Differences between startScan() and scan()

### DISCUSSION THREAD

**Discussion by randy-johnson** [capawesome-team/capacitor-mlkit#62] [Q&A] [answered]
Hello,

I am wondering what the differences are between startScan() and scan() and why the need for both?

Thanks,
Randy

**Comment by robingenz** [comment, marked as answer]
Please take a look at the [documentation](https://capawesome.io/plugins/mlkit/barcode-scanning/).

`startScan(...)` starts the scan and returns detected barcodes via the `barcodeScanned` listener. Multiple barcodes can be scanned in one session.
`scan(...)` scans a barcode with a ready-to-use interface which means you don't need any WebView customization. The scanned barcode is returned as the result of the method.

**Reply by randy-johnson** [reply]
Hi @robingenz ,  Thank You. I did look at the documentation but could not determine the difference.  Can you point me to the spot in the docs?

**Reply by robingenz** [reply]
1. [`startScan(...)`](https://capawesome.io/plugins/mlkit/barcode-scanning/#startscan)
2. [`scan(...)`](https://capawesome.io/plugins/mlkit/barcode-scanning/#scan)
